src/manifold/src/smoothing.cpp

```diff
@@ -947,18 +947,17 @@ void Manifold::Impl::CreateTangents(std::vector<Smoothness> sharpenedEdges) {
       tangent[second] = CircularTangent(
           -newTangent, vertPos_[halfedge_[second].endVert] - pos);
 
-      auto SmoothHalf = [&](int first, int last, float smoothness) {
-        int current = NextHalfedge(halfedge_[first].pairedHalfedge);
-        while (current != last) {
-          tangent[current] = smoothness * tangent[current];
-          current = NextHalfedge(halfedge_[current].pairedHalfedge);
-        }
-      };
-
-      SmoothHalf(first, second,
-                 (vert[0].second.smoothness + vert[1].first.smoothness) / 2);
-      SmoothHalf(second, first,
-                 (vert[1].second.smoothness + vert[0].first.smoothness) / 2);
+      float smoothness =
```

Comment:
should we run `ForVert` twice here?

Reply:
No, I changed the function a bit so it could finish in one pass instead.